conf_parse.go

```diff
@@ -0,0 +1,243 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package questdb
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type configData struct {
+	Schema        string
+	KeyValuePairs map[string]string
+}
+
+func confFromStr(conf string) (*lineSenderConfig, error) {
+	senderConf := &lineSenderConfig{}
+
+	data, err := parseConfigStr(conf)
+	if err != nil {
+		return nil, err
+	}
+
+	switch data.Schema {
+	case "http":
+		senderConf.senderType = httpSenderType
+	case "https":
+		senderConf.senderType = httpSenderType
+		senderConf.tlsMode = tlsEnabled
+	case "tcp":
+		senderConf.senderType = tcpSenderType
+	case "tcps":
+		senderConf.senderType = tcpSenderType
+		senderConf.tlsMode = tlsEnabled
+	default:
+		return nil, fmt.Errorf("invalid schema: %s", data.Schema)
+	}
+
+	for k, v := range data.KeyValuePairs {
+		switch strings.ToLower(k) {
+		case "addr":
+			senderConf.address = v
+		case "user":
```

Comment:
@puzpuzpuz @amunra  is it "user" or "username"?

Reply:
It's "user"

Reply:
It's "user"

Reply:
The python docs have "username"

```
from questdb.ingress import Sender, ServerTimestamp

conf = (
    'tcp::addr=localhost:9009;' +
    'username=testUser1;' +
    'token=5UjEMuA0Pj5pjK8a-fa24dyIf-Es5mYny3oE_Wmus48;' +

...
```

Reply:
Hmm, same with Rust client. Let's change it to username in Go client too.

cc @jerrinot since Java sender expects "user", not "username".

Reply:
Updated in 40f560a4233821f7f2b5377650e948b52ce3986c